Source/ProgressionSystemRuntime/Private/LevelActors/PSStarActor.cpp

```diff
@@ -45,44 +59,96 @@ void APSStarActor::OnLocalCharacterReady(APlayerCharacter* Character, int32 Char
 	if (ensureMsgf(LocalPC, TEXT("ASSERT: [%i] %hs:\n'LocalPC' is null!"), __LINE__, __FUNCTION__))
 	{
 		LocalPC->OnAnyCinematicStarted.AddDynamic(this, &APSStarActor::OnAnyCinematicStarted);
+		Character->OnPlayerTypeChanged.AddUniqueDynamic(this, &ThisClass::OnPlayerTypeChanged);
 	}
 }
 
+// Called when the current  player was changed
+void APSStarActor::OnPlayerTypeChanged(FPlayerTag PlayerTag)
+{
+	InitStarMenuAnimation();
+}
+
+// Called when the current game state was changed
+void APSStarActor::OnGameStateChanged(ECurrentGameState GameState)
+{
+	InitStarMenuAnimation();
```

Comment:
Please check is it Menu state. Right not it activates for each state like GameStarting, InGame etc

Reply:
done as part of: https://github.com/h4rdmol/ProgressionSystem/pull/14/commits/6141759e381d2b8658971efac878aaebce1a9cf7